### PR TITLE
Remove the AWS XRay sampler code

### PR DESCRIPTION
**Pull request by jkwatson**
These have been moved over to the opentelemetry-java-contrib project.

### CODE REVIEW

File: sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/trace/AwsXrayIdGenerator.java [hunk 1]
```diff
@@ -19,7 +19,10 @@
  * @see <a
  *     href="https://docs.aws.amazon.com/xray/latest/devguide/xray-api-sendingdata.html#xray-api-traceids">Generating
  *     Trace IDs</a>
+ * @deprecated Please use the AwsXrayIdGenerator from the
+ *     https://github.com/open-telemetry/opentelemetry-java-contrib project.
  */
+@Deprecated
```
Comment (anuraaga) [inline sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/trace/AwsXrayIdGenerator.java:25]: Can we deprecate in next release so it's after contrib c artifact of published?

Comment (jkwatson) [reply sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/trace/AwsXrayIdGenerator.java:25]: I'd be fine with that. Happy to revert that bit.

Comment (jkwatson) [reply sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/trace/AwsXrayIdGenerator.java]: done

**Review by anuraaga — APPROVED** [review]
I missed my opportunity to send a fun deletion PR oh well :)

**Comment by jkwatson** [comment]
> I missed my opportunity to send a fun deletion PR oh well :)

you can still merge it if you want!  :)

**Pull request merged**